api/v1/endpoints/listings.py

```diff
@@ -58,24 +64,29 @@ def post(self, listing_hash):
             md5_sum = request.form.get('md5_sum')
         if request.form.get('tags'):
             tags = request.form.get('tags')
-        for item in request.files.items():
+        if request.form.get('filenames'):
+            filenames = request.form.get('filenames').split(',')
+        for idx, item in enumerate(request.files.items()):
             destination = os.path.join('/tmp/uploads/')
-            log.info(f'Saving {item[0]} to {destination}')
+            filename = filenames[idx] if idx < len(filenames) else item[0]
+            log.info(f'Saving {filename} to {destination}')
             if not os.path.exists(destination):
                 os.makedirs(destination)
-            item[1].save(f'{destination}{item[0]}')
-            with open(f'{destination}{item[0]}', 'rb') as data:
+            item[1].save(f'{destination}{filename}')
+            with open(f'{destination}{filename}', 'rb') as data:
                 contents = data.read()
                 uploaded_md5 = hashlib.md5(contents).hexdigest()
             if uploaded_md5 != md5_sum:
                 return 'File upload failed', 500
             local_finish = time.time()
             timings['local_save'] = local_finish - start_time
-            log.info(f'Saving {item[0]} to S3 bucket ffa-dev')
+            log.info(f'Saving {filename} to S3 bucket ffa-dev')
             s3 = boto3.client('s3')
-            with open(f'{destination}{item[0]}', 'rb') as data:
-                s3.upload_fileobj(data, 'ffa-dev', item[0])
+            with open(f'{destination}{filename}', 'rb') as data:
+                # apparently this overwrites existing files.
```

Comment:
Yeah, true. Maybe subfolder it by public address, like `{destination}/{publicaddress}{filename}.

Reply:
Yeah just pointing it out.  Can resolve later.